src/jimgw/transforms.py

```diff
@@ -0,0 +1,177 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+import jax
+import jax.numpy as jnp
+from chex import assert_rank
+from jaxtyping import Float
+
+
+class Transform(ABC):
+    """
+    Base class for transform.
+
+    The idea of transform should be used on distribtuion,
+    """
+
+    name_mapping: tuple[list[str], list[str]]
+    transform_func: Callable[[dict[str, Float]], dict[str, Float]]
```

Comment:
Hi @kazewong , here, we make transform_func an attribute of Transform, but it is not given as an argument to __init__, why is that the case? To me it would make sense if the user should always give a transform func as well, otherwise, that might be in conflict with the purpose of the transform?

Reply:
I don't think transform_func should be an input to the class because this potentially encourages the user to put runtime unsafe function into the input, which will cause potential instability in the code. For example, if one uses `lambda x: arccos(x)` as a `transform_func`, the transform function will actually return nan because of its negative gradient and we are taking the log of it. Whenever the user wants to define a new transform, they should also be mindful about its properties and interaction with the rest of the code. For this reason, I don't think we should have `transform_func` as an input to init.

The typing hint in `transform_func` seems to be an mistake, will correct it.